gw_eccentricity/eccDefinition.py

```diff
@@ -579,6 +617,11 @@ def process_data_dict(self,
                 "1. 'hlm' OR \n"
                 "2. 'amplm' and 'phaselm'\n"
                 "But not both 1. and 2. at the same time."))
+        # if the system is precessing and the modes are given in the inertial
+        # frame, rotate the modes and obtain the corresponding modes in the
+        # coprecessing frame
+        if self.precessing is True and self.frame == "inertial":
+            dataDict = self.rotate_modes(dataDict)
```

Comment:
Once you do this, self.frame will be inertial, but dataDict will be in coprecessing?
Seeing as you do want to keep frame in self, why not update it at this point?
Also, `rotate_modes` seems too generic, how about `transform_inertial_to_coprecessing`?

Reply:
Done